File: ds_6_string/L696.cpp
//
// Created by 李恒 on 2020/7/1.
//
#include <iostream>
#include <cstring>

using namespace std;
class Solution {
public:
    int countBinarySubstrings(string s) {
        int preLen = 0, curLen = 1, count = 0;
        for (int i = 1; i < s.size(); i++) {
            if (s[i] == s[i-1]) {
                curLen++;
            } else {
                preLen = curLen;
                curLen = 1;
            }

            if (preLen >= curLen) {
                count++;
            }
        }
        return count;
    }
};

File: python-algorithm/ds_6_string/L696.py
# -*- coding: utf-8 -*-

# 这道题不太好想到解题办法
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        preLen = 0
        curLen = 1
        count = 0
        for i in range(1, len(s)) :
            if s[i] == s[i-1] :
                curLen += 1
            else :
                preLen = curLen
                curLen = 1
            if preLen >= curLen:
                count += 1
        return count